sample-compose/app/src/main/java/co/nimblehq/sample/compose/ui/AppDestination.kt

```diff
@@ -16,6 +18,10 @@ sealed class AppDestination(val route: String = "") {
 
     object Up : AppDestination()
 
+    object RootNavGraph : AppDestination("rootNavGraph")
```

Comment:
@luongvo How about we create a new Destination Class for `Main`? I think we should have only one `NavGraph` per destination file at the top. It is clearer and can know where to put the destination. Or we won't have consistency or grouping.

**With separate Destination file**
```
sealed class ADestination {
 object ANavGraph : AppDestination()
 object ScreenA : AppDestination()
 object ScreenB : AppDestination()
}
```

**Without separate Destination file**
```
sealed class AppDestination {
 object ANavGraph : AppDestination()
 object BNavGraph : AppDestination()
 object ScreenA : AppDestination()
 object CNavGraph : AppDestination()
 object ScreenB : AppDestination()
 object ScreenD : AppDestination()
}
```
What do you think?

Reply:
@kaungkhantsoe good idea 👍 
I refactored the AppDestination into 3 components:
- BaseDestination: base destination logic, such as route, arguments, destination, or Up action.
- AppDestination: all nav graphs, such as RootNavGraph, RootNavGraph
- MainDestination: each nav graph's destinations, such as Main and Home.

Updated in https://github.com/nimblehq/android-templates/pull/529/commits/1fc5e8b41786dab053d08e57d522ed46cf26cfab

Reply:
This also reflects the structure from the GT project cc @hoangnguyen92dn 💪